src/main/java/frc/robot/commands/autonomousCommands/AutonomousShooterCommand.java

```diff
@@ -57,27 +62,24 @@ public void initialize(){
     public void execute(){
         shooterSpeedController.setLaunchSpeed(ShooterMath.determineLaunchSpeed(distanceToGoal + INNER_DISTANCE_FROM_TARGET));
 
-            while(ballShot == false){
-                if(shooterSpeedController.isAtSpeed()){
-                if(distanceSensor.isPowerCellLoaded()){
-                    indexerController.intake();
-                    hopperController.stop();
-                    while(distanceSensor.isPowerCellLoaded()){
-                        //keep running indexer
-                    }
-                    startTime = Timer.getFPGATimestamp();
-                    while(Timer.getFPGATimestamp()< startTime + 1.0){
-                        //keep indexer running for one second
-                    }
-                    ballsShot ++;
-                }
-                else{
-                    indexerController.intake();
-                    hopperController.start();
-                }
-                }
+        if(shooterSpeedController.isAtSpeed() && Timer.getFPGATimestamp() - startTime > LAUNCH_TIME ){// if the shooter is at speed and has given time for last ball to shoot
+            indexerIntakeCommand.schedule();
+            if(distanceSensor.isPowerCellLoaded() && !(ballinIndexer == false)){
```

Comment:
!(ballinIndexer == false) is essentially ballinIndexer==true.

! is a unary operator (unary as in it takes one argument), and it flips it from true to false or false to true.

What you actually want here is just !ballinIndexer

Reply:
The other type of operator is called a binary operator, which takes two arguments and does something with them.

Some Examples are +,-,*,/,%,<,>,<=,>=,==,!=,||,&&

Reply:
fixed